### ISSUE TITLE
Long-Time White-Screen displays everytime app starts, and it's confusing !! :\ 

### ISSUE THREAD

**Issue by tomavic** [ionic-team/ionic#14419]
Since Ionic 2 rise up, and with releasing Ionic 3, I noticed after building the app to Android or iOS and installing it on device properly, I found that there is a White screen showing up for like 4 or 6 seconds, then comes the splash screen and the rest of the app.

It's been so hard for me to convince the clients that they have to accept the app as it is, although that issue was NOT exists in Ionic 1.x !!

Previously, In Ionic 1.x when you start plain project form scratch and build it (for example Android), installing it to device, and If you open it you will meet the splash screen then the rest of the app etc...
and that was Awesome 

Please could you look into this and take care of it in Ionic 4.x series ASAP :)

**Ionic version:**  
[ ] **2.x**
[*] **3.x**
[ ] **4.x**

**I'm submitting a ...**  actually I don't know if it's really an issue need to be fixed, or a feature to be requested !
[*] bug report
[* ] feature request

**Current behavior:**
White screen showing up after opening the apk, for like 3-6 seconds

**Expected behavior:**
NOT displaying that white screen, or at least decreasing it's time to be less than 1000ms and not more than that. That would be the worst case btw

**Other information:**
Even using CrossWalk doesn't make any difference

**Ionic info:** 

```
cli packages: (C:\Users\Hatem\AppData\Roaming\npm\node_modules)

    @ionic/cli-utils  : 1.19.1
    ionic (Ionic CLI) : 3.19.1

global packages:

    cordova (Cordova CLI) : 8.0.0

local packages:

    @ionic/app-scripts : 3.1.8
    Cordova Platforms  : none
    Ionic Framework    : ionic-angular 3.9.2

System:

    Android SDK Tools : 26.1.1
    Node              : v8.10.0
    npm               : 5.5.1
    OS                : Windows 10

Environment Variables:

    ANDROID_HOME : C:\Users\Hatem\AppData\Local\Android\sdk1

Misc:

    backend : pro

```

**Comment by newuser44** [comment]
I have not seen an app have a white screen BEFORE the splash screen.  
This happens on both ios and android?

I assume your config.xml has the normal splash screen settings?

**Comment by tomavic** [comment]
@newuser44 

You could start new project by running `ionic start myApp tabs` 
then build it to Android or iOS
then run it on your device Android/iOS
you'll find what I am talking about exaclty
A white screen lasts for 5 seconds, then splash screen, then next page whatever it is

**Comment by kensodemann** [comment]
@tomavic - I have not seen that either and I have several test apps built in a similar fashion as to what you have stated in your above comment just due to testing things.

What I have noticed with regard to this is that on some cheaper Android devices there will be a period of a grey screen between the splash and the app, but never before the splash.

Here is a link showing what I see when launching an Ionic app from my phone: https://drive.google.com/file/d/1-D79urp1Gfhtruxr7RQ8x-mF7Rcr3Dc2/view?usp=sharing

I launch the app around second 2 and it goes immediately to the splash screen, after which it goes right into the app.

Could you perhaps share a GitHub repo to an app that exhibits the behavior you are talking about?

Perhaps there is a difference between how I built it and you built it? The build I have is not a production build (which if anything should make any performance issue worse) and is side-loaded on the device from Xcode.

**Comment by tomavic** [comment]
Hi @kensodemann 

Thanks for your comment and the video. 

What I am talking about exactly happens in Android Devices (even the cheaper one which have android less than 4.2, which Ionic 3.x doesn't support and I have to use cross-walk to build the app).

What happens is a White screen lasts for 5 sec After splash and not before it. 

There is nothing special about the apk itself. I run `ionic cordova build android` and installed the apk

Please checkout this video:
https://youtu.be/Pf__0R3NYxY

**Comment by newuser44** [comment]
Up your SplashScreenDelay time and see what happens.
I would also do a build with --prod.     You will have to change your environment.ts url to where you want it to point.

**Comment by tomavic** [comment]
@newuser44 I had increased splash screen delay time to be 6000, and the white screen is not visible anymore.
`<preference name="SplashScreenDelay" value="6000" />`


But the problem is -and that what I meant in first place- the device take too much time to be ready, and I hope Ionic team will take care of this in the next Major release to decrease that time

**Comment by newuser44** [comment]
This is part of Angular.  It has two ways of compiling code,  JIT  or AOT, I think I posted a link to this above.  

If you build your code with --prod, it switched over to AOT and the load time is WAY quicker.  This is how your should build  release anyways.   

We went through the same thing.  Our android app did not have a splash screen before,  moved over to Ionic,  started to see how long the Dev build was, ended up adding the splash screen.  With the dev build it was close to 18 seconds to load,  with the --prod build (AOT) its just under 4 seconds.
